### PR TITLE
Move binaries to structopt, remove some extern crate-s

### PR DESCRIPTION
**Pull request by pzmarzly**
structopt panics at runtime (and it is clap-based), but the utils seems to work fine. I hope there aren't any copy-paste mistakes here.

### CODE REVIEW

**Comment by realcr** [comment]
@pzmarzly : Thanks! I hope to finish reviewing this PR today.

> structopt panics at runtime

Why do you mean by panics at runtime? Is this a bug? If so, how can we reproduce it?

File: components/bin/src/bin/stmgr.rs [hunk 1]
```diff
@@ -43,26 +43,124 @@ enum InitNodeDbError {
...
+
+#[derive(Debug, StructOpt)]
+struct AppTicketCmd {
+    /// StCtrl app identity file path
+    #[structopt(parse(from_os_str), short = "i", long = "idfile")]
+    idfile: PathBuf,
+    /// Application ticket output file path
+    #[structopt(parse(from_os_str), short = "o", long = "output")]
+    output: PathBuf,
+    /// Permission to request reports
```
Comment (realcr) [inline components/bin/src/bin/stmgr.rs]: Comment should be: "Permission to request routes"

File: components/proto/src/file/app.rs [hunk 2]
```diff
@@ -90,16 +90,32 @@ pub fn store_trusted_app_to_file(
     Ok(())
 }
 
+#[derive(Debug)]
+pub enum AppDirectoryError {
```
Comment (realcr) [inline components/proto/src/file/app.rs]: Good call here. The previous error didn't indicate which file was problematic, right?

**Comment by realcr** [comment]
@pzmarzly : Great work here. The command line util's code looks much better. Alsok, thanks for the 2018 edition improvements.
I'm merging this. If any problems occurred during the migration to structopt future tests will notify us.

**Comment by pzmarzly** [comment]
Structopt panics when there are 2 conflicting flags, e.g. 2 parameters sharing the same `short`. But from what I noticed, even `--help` panics, and I think it also does that when the conflict is in subcommand that you aren't currently invoking.

File: components/bin/src/bin/stmgr.rs (hunk 1 above)
Comment (pzmarzly) [reply components/bin/src/bin/stmgr.rs]: Could you change it? This PR is already merged, I don't think it's worth it for me to make another one.

Comment (realcr) [reply components/bin/src/bin/stmgr.rs]: Done on PR: https://github.com/freedomlayer/offst/pull/176

**Comment by realcr** [comment]
Interesting. I will check if this problem occurs later in some test program. I thought that the panic occurs only if two substructs are flattened, and not for two distinct subcommands in an enum.

**Comment by pzmarzly** [comment]
If two subcommands share same shortflag, it's okay. If you are running `my_bin subcommand1 --help` when the conflict is in `my_bin subcommand2`, I think it still panics (imo it should)

**Pull request merged**